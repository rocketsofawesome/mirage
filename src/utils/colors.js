const colorFromProp = (propName) => (props) => {
  const propValue = props[propName]
  return props.theme.colors[propValue] || propValue
}

export default colorFromProp
