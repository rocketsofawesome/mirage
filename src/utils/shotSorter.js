import React, { Component } from 'react'
import { Layer } from 'cloudinary-core'

import cloudinary from 'services/cloudinary'

export const shotTypeSortOrder = [
  'front',
  'on_model',
  'back',
  'detail',
  'other'
]

export function sortShots(shots) {
  return shots.sort((shot1, shot2) =>
    shotTypeSortOrder.indexOf(shot1.shot_type) - shotTypeSortOrder.indexOf(shot2.shot_type)
  )
}

function shotUrl(shot) {
  const config = {
    transformation: 'plp_product_shot',
    format: 'jpg'
  }

  if (shot.badge) {
    const layer = new Layer({ publicId: shot.badge.public_id })
    config.overlay = layer
    config.width = 225
    config.gravity = 'north_east'
    config.x = 20
    config.y = 20
  }

  return cloudinary.url(shot.cloudinary_key, config)
}

export const withSortedShots = (WrappedComponent) => {
  return class extends Component {
    render () {
      const { product, shots } = this.props
      const sortedShots = sortShots(shots)
      const images = sortedShots.map(shot => ({
        alt: `${product.name} ${shot.shot_type}`,
        src: shotUrl(shot)
      }))
      return <WrappedComponent {...this.props} images={images} />
    }
  }
}
