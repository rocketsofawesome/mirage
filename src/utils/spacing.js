export const spacers = {
  none: 0,
  xs: '0.25rem',
  sm: '0.5rem',
  md: '1rem',
  lg: '1.5rem',
  xl: '3rem'
}

export const spaceFromProp = (propName) => (props) => {
  const propValue = props[propName]
  return spacers[propValue]
}
