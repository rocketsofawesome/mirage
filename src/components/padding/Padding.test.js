import React from 'react'
import 'jest-styled-components'
import { Padding } from 'SRC'
import { spacers } from 'utils/spacing'

const { shallowWithTheme } = global

const createPadding = props => {
  return shallowWithTheme(<Padding {...props} />)
}

describe('(StyledComponent) Padding', () => {
  test('matches the snapshot', () => {
    expect(createPadding()).toMatchSnapshot()
  })

  test('has no padding if props are not provided', () => {
    const component = createPadding()
    expect(component).toHaveStyleRule({ 'padding-top': 0 })
    expect(component).toHaveStyleRule({ 'padding-bottom': 0 })
    expect(component).toHaveStyleRule({ 'padding-right': 0 })
    expect(component).toHaveStyleRule({ 'padding-left': 0 })
  })

  test('applies appropriate padding with qualitative values', () => {
    const component = createPadding({ t: 'lg', b: 'md', r: 'sm', l: 'xl' })
    expect(component).toHaveStyleRule({ 'padding-top': spacers.lg })
    expect(component).toHaveStyleRule({ 'padding-bottom': spacers.md })
    expect(component).toHaveStyleRule({ 'padding-right': spacers.sm })
    expect(component).toHaveStyleRule({ 'padding-left': spacers.xl })
  })

  test('applies appropriate padding with quantitative values', () => {
    const component = createPadding({
      t: '6rem',
      b: '2rem',
      r: '10rem',
      l: '5rem'
    })
    expect(component).toHaveStyleRule({
      'padding-top': '6rem'
    })
    expect(component).toHaveStyleRule({
      'padding-bottom': '2rem'
    })
    expect(component).toHaveStyleRule({
      'padding-right': '10rem'
    })
    expect(component).toHaveStyleRule({
      'padding-feft': '5rem'
    })
  })
})
