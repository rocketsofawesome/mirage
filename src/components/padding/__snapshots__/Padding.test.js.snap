// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(StyledComponent) Padding matches the snapshot 1`] = `
.c0 {
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 0;
  padding-right: 0;
}

<div
  className="c0"
  x={0}
  y={0}
/>
`;
